### PR TITLE
docs: update instructions for scaling down store-gateways

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
If we don't remove the instances form the ring before proceeding to the
next availability zone, then we risk losing redundancy for the blocks
that are replicated on these instances.

This PR updates the docs to mention forgetting the instances in the
store-gateway ring.

Signed-off-by: Dimitar Dimitrov <dimitar.dimitrov@grafana.com>

### CODE REVIEW

File: docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md [hunk 1]
```diff
@@ -136,7 +136,12 @@ The required amount of time to wait depends on your configuration and it's the m
 
 To guarantee no downtime when scaling down [store-gateways]({{< relref "../architecture/components/store-gateway.md" >}}), complete the following steps:
 
-1. Scale down no more than two store-gateways at the same time.
 1. Ensure at least `-store-gateway.sharding-ring.replication-factor` store-gateway instances are running (three when running Grafana Mimir with the default configuration).
-
-> **Note:** If you enabled [zone-aware replication]({{< relref "../configure/configure-zone-aware-replication.md" >}}) for store-gateways, you can in parallel scale down any number of store-gateway instances in one zone at a time.
+1. Scale down no more than two store-gateways at the same time. If you enabled [zone-aware replication]({{< relref "../configure/configure-zone-aware-replication.md" >}})
```
Comment (osg-grafana) [inline docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md:140]:
```suggestion
1. If you enabled [zone-aware replication]({{< relref "../configure/configure-zone-aware-replication.md" >}}), scale down no more than two store-gateways at the same time.
```
Rationale: save the reader a tad of time if they do no have zone-aware replication enabled. (They can stop reading and go on to the next step.)

File: docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md [hunk 2]
```diff
@@ -136,7 +136,12 @@ The required amount of time to wait depends on your configuration and it's the m
...
+   for store-gateways, you can in parallel scale down any number of store-gateway instances in one zone at a time. Zone-aware replication is enabled by default in the `mimir-distributed` Helm chart.
```
Comment (osg-grafana) [inline docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md:141]:
```suggestion
1. Scale down no more than two store-gateways at the same time.
  If you enabled [zone-aware replication]({{< relref "../configure/configure-zone-aware-replication.md" >}})
   for store-gateways, you can in parallel scale down any number of store-gateway instances in one zone at a time.
   Zone-aware replication is enabled by default in the `mimir-distributed` Helm chart.
```
Rationale: Separate instruction from additional information or context.

**Review by aknuds1 — APPROVED** [review]

File: docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md [hunk 3]
```diff
@@ -136,7 +136,12 @@ The required amount of time to wait depends on your configuration and it's the m
...
+1. Stop the store-gateway instances you want to scale down.
+1. Remove the stopped instances from the store-gateway ring. In your browser go to the `GET /store-gateway/ring` page that store-gateways expose on their HTTP port
```
Comment (pracucci) [inline docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md:143]: This depends on this setting:

```
# Unregister from the ring upon clean shutdown.
  # CLI flag: -store-gateway.sharding-ring.unregister-on-shutdown
  [unregister_on_shutdown: <boolean> | default = true]
```

With the default config, it unregisters from the ring on shutdown so there's no action to take. I would clarify it.

File: docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md [hunk 4]
```diff
@@ -136,7 +136,12 @@ The required amount of time to wait depends on your configuration and it's the m
...
+   and click "Forget" on the instances that you scaled down.
+   Alternatively, you can wait for 10 times the value of `-store-gateway.sharding-ring.heartbeat-timeout` (1m by default)
+   for the store-gateways to be automatically forgotten from the ring.
+1. Proceed with the next two store-gateway replicas or the next zone when using zone-aware replication.
```
Comment (osg-grafana) [inline docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md:149]:
```suggestion
1. If you are using zone-aware replication, proceed with the next two store-gateway replicas or the next zone.
```

Comment (osg-grafana) [reply docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md:149]: In line 141 (https://github.com/grafana/mimir/pull/3698/files#diff-b2b4912b253f31f2422ed4f97255308f7b339700558844adf1ea0965a70292f6R141), there is an "if" and here there is a "when". Does "when" mean "if" or "while" here?

Comment (dimitarvdimitrov) [reply docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md:149]:
```suggestion
1. Proceed with the next two store-gateway replicas. If you are using zone-aware replication, the proceed with the next zone.
```

File: docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md [hunk 5]
```diff
@@ -136,7 +136,12 @@ The required amount of time to wait depends on your configuration and it's the m
...
-
-> **Note:** If you enabled [zone-aware replication]({{< relref "../configure/configure-zone-aware-replication.md" >}}) for store-gateways, you can in parallel scale down any number of store-gateway instances in one zone at a time.
+1. Scale down no more than two store-gateways at the same time. If you enabled [zone-aware replication]({{< relref "../configure/configure-zone-aware-replication.md" >}})
+   for store-gateways, you can in parallel scale down any number of store-gateway instances in one zone at a time. Zone-aware replication is enabled by default in the `mimir-distributed` Helm chart.
+1. Stop the store-gateway instances you want to scale down.
+1. Remove the stopped instances from the store-gateway ring. In your browser go to the `GET /store-gateway/ring` page that store-gateways expose on their HTTP port
+   and click "Forget" on the instances that you scaled down.
+   Alternatively, you can wait for 10 times the value of `-store-gateway.sharding-ring.heartbeat-timeout` (1m by default)
+   for the store-gateways to be automatically forgotten from the ring.
```
[the remark this answers is not in the document]
Comment (osg-grafana) [reply docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md:148]: As I re-read this, I do not like my initial suggestion and recommend the second suggestion.

**Review by osg-grafana — APPROVED** [review]
Unblocking with feedback; if you want me to take another look, lemme know.

File: docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (dimitarvdimitrov) [reply docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md]: i think i addressed this in https://github.com/grafana/mimir/pull/3698/commits/d8b3b96a738fb51838bfdb205d881349462cc103

i went with the second suggestion

File: docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md (hunk 3 above)
Comment (dimitarvdimitrov) [reply docs/sources/mimir/operators-guide/run-production-environment/scaling-out.md]: i think i addressed this in https://github.com/grafana/mimir/pull/3698/commits/d8b3b96a738fb51838bfdb205d881349462cc103

**Review by pracucci — APPROVED** [review]
LGTM, thanks!

**Pull request merged**